### ISSUE TITLE
Merge Booking Cells for the same booking.

### ISSUE THREAD

**Issue by bytefactory73** [tkrebs/ep3-bs#288]
In our local installation I improved the look of the bookings to join the cells vertically that belong to the same booking.

before:

![image](https://user-images.githubusercontent.com/9210200/80943012-7d282980-8de6-11ea-8cf4-93236f1f0e85.png)

after:

![image](https://user-images.githubusercontent.com/9210200/80942712-c461ea80-8de5-11ea-8bc5-176031a98203.png)

If the owner if this project likes this i could bring that change over to the master branch.

**Comment by bytefactory73** [comment]
required style changes:
https://github.com/bytefactory73/ep3-bs/commit/6c55069abdb99df76eb67687ad47425d6a1f5053
code changes:
https://github.com/bytefactory73/ep3-bs/commit/e63d3da93c6e40d439e3cb414be07e5dbbcff73f

**Comment by tkrebs** [comment]
Thank you for sharing!

I will look into the code changes soon.

Meanwhile, I have pinned the issue so other users can implement this on demand.

**Comment by bytefactory73** [comment]
It's a bit hacky because i did not understand your data structures (i'm coming from c++ and php is not really my language :) - it's also hard for me to debug stuff because i can only run it on the server without any debugger).

so i made it at the central place where the cells are output. i check for group-id-xxxx and merge all cells with the same group id. that did the trick for me.

to join the columns you need to leave out the cells in the other rows and span the col with rowspan="<cnt>" over the other rows.

**Comment by tkrebs** [comment]
Your approach looks fine to me on first impression. Might have to look deeper though. ;)

Debugging PHP (with Xdebug) is indeed not the easiest thing to setup, but once it's working, it's actually quite pleasant.

**Comment by bytefactory73** [comment]
i added another fix for the stylesheet ontop of this changes:
https://github.com/bytefactory73/ep3-bs/commit/94572c1e6c9ec3d8f260f5f22f4c27317fb67758

**Comment by gitroli** [comment]
hi I already had this request and we solved it

**Comment by gitroli** [comment]
siehe hier:
https://github.com/hschiebold/ep3-bs/issues/1

ist zwar fork von dir aber das issue bei dir finde ich gerade nicht mehr

**Comment by Shineson1001** [comment]
Hi,
ich habe mir die 3 geänderten Dateien von @bytefactory73 eingespielt ... sieht seht gut. Besten Dank.

**Comment by bytefactory73** [comment]
@all leider hatten meine änderungen noch negative auswirkungen auf folgende fälle:

- wenn alle plätze eines tages buchungen über mehrere stunden hatten wurden zeilen zu viele zeilen entfernt.
- events über alle plätze (über mehrere spalten) haben nicht mehr funktioniert.

langsam wird es auch unübersichtlich mit zu vielen einzel submits.
ich werde mal einen branch von tkrebs/develop erzeugen und dort nur diese changes einbringen ...

ich melde mich, sobald das fertig ist.

**Comment by Shineson1001** [comment]
Hi @bytefactory73,
ich habe die Änderungen erst mal rückgängig gemacht ... Die Zuordnung zwischen Uhrzeit, Platz und Buchungen verschiebt sich
![grafik](https://user-images.githubusercontent.com/22273771/81332072-cad0ba80-90a2-11ea-9053-a5c962e97565.png)

**Comment by bytefactory73** [comment]
genau, das war das problem. meine nachfolgenden fixes beheben das problem ...

**Comment by Rudimo76** [comment]
@bytefactory73 : welche nachfolgenden fixes? Bei uns ist nämlich nun das selbe Problem der Verschiebung aufgetreten.

**Comment by Shineson1001** [comment]
Hi @Rudimo76, 
ich bin erst mal den Weg von @gitroli gegangen ... hier werden die Zellen nicht serverseitig per php verbunden, sondern clientseitig per JavaScript.

Dafür benötigt man eine angepasste index.min.js im Ordner /public/js/controller/calendar/
https://pastebin.com/t111QwYG

**Comment by bytefactory73** [comment]
gute nachrichten:

ich habe nun alle relevanten änderungen für meine methode von org/master in 

https://github.com/bytefactory73/ep3-bs/tree/dev/cellmerge2.0

commited. 

wenn ihr diesen branch zu euch mergt solltet habt ihr alles zusammen.

**Comment by bytefactory73** [comment]
> Hi @bytefactory73,
> ich habe die Änderungen erst mal rückgängig gemacht ... Die Zuordnung zwischen Uhrzeit, Platz und Buchungen verschiebt sich
> ![grafik](https://user-images.githubusercontent.com/22273771/81332072-cad0ba80-90a2-11ea-9053-a5c962e97565.png)

fixed in my latest changes

**Comment by Shineson1001** [comment]
Hi @bytefactory73, 
ja, sieht gut aus

![grafik](https://user-images.githubusercontent.com/22273771/81498449-588bf000-92c5-11ea-88d4-5a35910c183c.png)

**Comment by JonesHH** [comment]
Hi,
in der 1.6.4 von Tobias scheint das ja implementiert.
Welche Dateien sind denn für diese Änderung angepasst worden?
Ich brauche wahrscheinlich die ursprüngliche Form.

gruss
jones

**Comment by Shineson1001** [comment]
Hi @JonesHH,
nee, die Anpassungen für die Zellen-Zusammenfassung ist noch nicht drin.

sieht also noch "ursprünglich" aus
![grafik](https://user-images.githubusercontent.com/22273771/84888182-312ffc00-b097-11ea-8e7c-c5aa4c8449d7.png)

**Comment by JonesHH** [comment]
Komisch, bei mir sieht es so aus
![image](https://user-images.githubusercontent.com/35833331/84908392-b2e25280-b0b4-11ea-9120-2870b63eb12d.png)

frische Installation der 1.6.4 auf ner Testumgebung

**Comment by Shineson1001** [comment]
Hi @JonesHH,
das betrifft aber nur die Events/Veranstaltungen ... die normalen Buchungen und Abos werden nicht als eine Zelle dargestellt.

Und die Zellen-Verbindungstechnik wird clientseitig gemacht ... nicht serverseitig, wie hier von @bytefactory73 vorgestellt Lösung.

**Comment by bytefactory73** [comment]
genau, die events wurden im originalcode per JS drübergepatched. dabei entstand ein flackern, da über den original zellen noch ein weiteres html element gelegt wurde.

das ist bei meinen änderungen nicht mehr so. es kommt quasi immer eine gültige html tabelle raus mit korrektem spanning.

**Comment by JonesHH** [comment]
OK, dachte immer, events und booking werden annähernd 
gleich behandelt.
Man lernt nie aus. Jetzt weiss ich, wo ich schauen muss.
Danke euch beiden

**Comment by DanielM1991** [comment]
Servus,
Sorry ich werde aus den ganzen Kommentaren leider nicht schlau. Kann mir jemand den aktuellen Stand mitteilen ob das Mergen nun funktioniert oder nicht? 
Ist es egal, wenn man die vertikale Zeitschiene gewählt hat?

Danke schon mal!

**Comment by cbits68** [comment]
Nachdem ich die Anpassungen aus cellmerge2.0 Tree eingespielt habe, sind die Hintergründe der zusammengeführten Zellen leider nicht mehr flächendeckend über die Buchung. Jemand eine Idee, woran das liegen könnte?
![test](https://user-images.githubusercontent.com/60112941/88803620-07e1b000-d1ad-11ea-88a1-bc883855d8b0.png)

**Comment by gitroli** [comment]
Hi ich habe den commit von bytefactory im Einsatz, funktioniert sehr gut, allerdings ist mir noch eine Kleinigkeit aufgefallen, ist ein Platz zB für 2h gebucht und klickt man darauf sieht die Popup Box so aus:
![grafik](https://user-images.githubusercontent.com/31999445/90484785-f6018600-e136-11ea-9cd0-63c989964697.png)
Also hier stimmt die Zeitangabe nicht.

**Comment by DanielM1991** [comment]
Funktioniert leider nicht im Landscape-mode :-(

#87
